[Unit]
Description=xl_auth daemon
After=syslog.target network.target

[Service]
User=xl_auth
Group=xl_auth
WorkingDirectory=/opt/xl_auth
Environment="PATH=/sbin:/bin:/usr/sbin:/usr/bin:/opt/xl_auth/venv/bin"
Environment="FLASK_APP=autoapp.py"
Environment="SERVER_NAME={{ xl_auth_server_name }}"
Environment="XL_AUTH_SECRET={{ xl_auth_secret | default('not_secret, as fallback for `vagrant up`') }}"
Environment="PREFERRED_URL_SCHEME=https"
Environment="FLASK_DEBUG=0"
Environment="SQLALCHEMY_DATABASE_URI=postgresql://xl_auth:xl_auth@localhost/prod"
Environment="OAUTH2_PROVIDER_TOKEN_EXPIRES_IN=3600"
ExecStart=/opt/xl_auth/venv/bin/flask prod_run
TimeoutStopSec=30

[Install]
WantedBy=multi-user.target
